packages/base/src/EventProvider.js

```diff
@@ -55,6 +54,17 @@ class EventProvider {
 		});
 	}
 
+	/**
+	 * Fires an event, awaits for all listeners' results to have resolved, and returns the results as an array.
```

Comment:
technically it doesn't await but returns the promises, could you change the description simply that it returns a promise and is always async?